### PR TITLE
Moved booking logic to calcom/features

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

Consolidates booking logic Needed to reuse on public API

refs calcom/cal.com#9126

## Type of change

- [x] Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

- [ ] yarn build
- [ ] yarn test
- [ ] yarn e2e

### CODE REVIEW

File: apps/web/components/webhook/WebhookDialogForm.tsx [hunk 1]
```diff
@@ -2,7 +2,8 @@ import { Webhook } from "@prisma/client";
 import { useEffect, useState } from "react";
 import { Controller, useForm } from "react-hook-form";
 
-import customTemplate, { hasTemplateIntegration } from "@calcom/features/webhooks/utils/integrationTemplate";
+import { WEBHOOK_TRIGGER_EVENTS_GROUPED_BY_APP } from "@calcom/features/webhooks/lib/constants";
+import customTemplate, { hasTemplateIntegration } from "@calcom/features/webhooks/lib/integrationTemplate";
 import { useLocale } from "@calcom/lib/hooks/useLocale";
```
Comment (zomars) [inline apps/web/components/webhook/WebhookDialogForm.tsx:7]: Consolidated to final location

File: apps/web/lib/webhooks/constants.ts [hunk 2]
```diff
@@ -1,18 +0,0 @@
-import { WebhookTriggerEvents } from "@prisma/client";
-
-// this is exported as we can't use `WebhookTriggerEvents` in the frontend straight-off
-
-export const WEBHOOK_TRIGGER_EVENTS_GROUPED_BY_APP = {
```
Comment (zomars) [inline apps/web/lib/webhooks/constants.ts:5]: deprecated

File: packages/app-store/ee/routing-forms/trpc-router.ts [hunk 3]
```diff
@@ -2,8 +2,8 @@ import { Prisma, WebhookTriggerEvents } from "@prisma/client";
 import { v4 as uuidv4 } from "uuid";
 import { z } from "zod";
 
-import getWebhooks from "@calcom/features/webhooks/utils/getWebhooks";
-import { sendGenericWebhookPayload } from "@calcom/lib/webhooks/sendPayload";
+import getWebhooks from "@calcom/features/webhooks/lib/getWebhooks";
+import { sendGenericWebhookPayload } from "@calcom/features/webhooks/lib/sendPayload";
```
Comment (zomars) [inline packages/app-store/ee/routing-forms/trpc-router.ts:6]: Moved to final location

**Review by zomars — COMMENTED** [review]
Ready for review

**Review by hariombalhara — APPROVED** [review]
Reviewed code. Looks good. Will test webhooks and booking

Edit: Looks good testing wise as well.

**Review by alishaz-polymath — APPROVED** [review]
Code looks good @zomars . I think this organization makes way more sense as we grow and long for an API flow consistent with our UI. 
Amazing as always 🚀

**Review by emrysal — APPROVED** [review]

**Pull request merged**